### PR TITLE
Optionally create name of aggregate expression from expressions

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/11707.

### CODE REVIEW

File: datafusion/core/src/physical_planner.rs [hunk 1]
```diff
@@ -1908,9 +1648,9 @@ pub fn create_aggregate_expr_and_maybe_filter(
 ) -> Result<AggregateExprWithOptionalArgs> {
     // unpack (nested) aliased logical expressions, e.g. "sum(col) as total"
     let (name, e) = match e {
-        Expr::Alias(Alias { expr, name, .. }) => (name.clone(), expr.as_ref()),
-        Expr::AggregateFunction(_) => (e.display_name().unwrap_or(physical_name(e)?), e),
-        _ => (physical_name(e)?, e),
+        Expr::Alias(Alias { expr, name, .. }) => (Some(name.clone()), expr.as_ref()),
+        Expr::AggregateFunction(_) => (None, e),
```
Comment (jayzhan211) [inline datafusion/core/src/physical_planner.rs:1652]: I think `e.display_name()` should be kept

Comment (lewiszlw) [reply datafusion/core/src/physical_planner.rs:1626]: Looks like `display_name()` contains filter but `physical_name()` not.

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by jayzhan211** [comment]
Thanks @lewiszlw

**Pull request merged**